### PR TITLE
Vulkan multisampling

### PR DESCRIPTION
**Pull request by apjagdale**
Enabling multisampling for vulkan.
Pipeline caching modified with sampleCount parameter.
Made vkImage parameterized for specifying tiling

GearVRf-DCO-1.0-Signed-off-by: Abhijit Jagdale a.jagdale@samsung.com

### CODE REVIEW

File: GVRf/Framework/framework/src/main/jni/vulkan/vk_imagebase.h [hunk 1]
```diff
@@ -31,25 +31,25 @@ namespace gvr {
...
 
 class vkImageBase
 {
     public:
-        explicit vkImageBase(VkImageViewType type) : outBuffer(new VkBuffer),imageType(type), size(0), format_(VK_FORMAT_R8G8B8A8_UNORM), usage_flags_(VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT),mSampleCount(1)
-        { }
-        explicit vkImageBase(VkImageViewType type, VkFormat format, int width, int height, int depth, VkImageUsageFlags flags, VkImageLayout imageLayout)
-        : mLayers(1),imageType(type), outBuffer(new VkBuffer), size(0), format_(format), usage_flags_(flags), width_(width), height_(height), depth_(depth), imageLayout(imageLayout), mSampleCount(1)
-        { }
+    vkImageBase(VkImageViewType type) : outBuffer(new VkBuffer),imageType(type), size(0), format_(VK_FORMAT_R8G8B8A8_UNORM), tiling_(VK_IMAGE_TILING_LINEAR), usage_flags_(VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT),mSampleCount(1)
```
Comment (roshanch) [inline GVRf/Framework/framework/src/main/jni/vulkan/vk_imagebase.h]: "explicit" removed, is it intentional?

Comment (apjagdale) [reply GVRf/Framework/framework/src/main/jni/vulkan/vk_imagebase.h]: Added it back.

**Comment by liaxim** [comment]
Who is going to review and verify this?

**Comment by roshanch** [comment]
I will do testing on this. Code looks okay to me.

**Comment by roshanch** [comment]
verified it with polyline and meshanimation sample. Good to go.

File: GVRf/Framework/framework/src/main/jni/vulkan/vk_render_texture.cpp [hunk 2]
```diff
@@ -45,10 +45,9 @@ const VkDescriptorImageInfo& VkRenderTexture::getDescriptorImage(){
 }
 
 void VkRenderTexture::createRenderPass(){
-    VulkanRenderer* vk_renderer= static_cast<VulkanRenderer*>(Renderer::getInstance());
-    VkRenderPass renderPass = vk_renderer->getCore()->createVkRenderPass(NORMAL_RENDERPASS, 1);
-
-    clear_values.resize(2);
+    VulkanRenderer* vk_renderer= reinterpret_cast<VulkanRenderer*>(Renderer::getInstance());
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/jni/vulkan/vk_render_texture.cpp]: Why back to reinterpret_cast?

**Comment by liaxim** [comment]
Planning to merge on Monday

**Pull request merged**